dlt/destinations/impl/filesystem/configuration.py

```diff
@@ -1,19 +1,56 @@
 import dataclasses
-from typing import Final, Type, Optional
 
+from typing import Final, List, Optional, Type
+
+from dlt.cli import echo as fmt
 from dlt.common.configuration import configspec, resolve_type
 from dlt.common.destination.reference import (
     CredentialsConfiguration,
     DestinationClientStagingConfiguration,
 )
+
 from dlt.common.storages import FilesystemConfiguration
+from dlt.destinations.impl.filesystem.typing import TCurrentDateTime, TExtraPlaceholders
+from pendulum import DateTime
+
+from dlt.destinations.path_utils import check_layout
 
 
 @configspec
 class FilesystemDestinationClientConfiguration(FilesystemConfiguration, DestinationClientStagingConfiguration):  # type: ignore[misc]
-    destination_type: Final[str] = dataclasses.field(default="filesystem", init=False, repr=False, compare=False)  # type: ignore
+    destination_type: Final[str] = dataclasses.field(  # type: ignore[misc]
+        default="filesystem", init=False, repr=False, compare=False
+    )
+    current_datetime: Optional[TCurrentDateTime] = None
+    extra_placeholders: Optional[TExtraPlaceholders] = None
 
     @resolve_type("credentials")
     def resolve_credentials_type(self) -> Type[CredentialsConfiguration]:
         # use known credentials or empty credentials for unknown protocol
-        return self.PROTOCOL_CREDENTIALS.get(self.protocol) or Optional[CredentialsConfiguration]  # type: ignore[return-value]
+        return (
+            self.PROTOCOL_CREDENTIALS.get(self.protocol)
+            or Optional[CredentialsConfiguration]  # type: ignore[return-value]
+        )
+
+    def on_resolved(self) -> None:
+        if callable(self.current_datetime):
```

Comment:
I think it is better not to call lambdas here. just check "statically" if all placeholders are there

Reply:
Should we then even check it because one might a function without typing thus return typpe will be NoneType?